src/huggingface_hub/constants.py

```diff
@@ -65,7 +65,7 @@ def _as_int(value: Optional[str]) -> Optional[int]:
 _HF_DEFAULT_STAGING_ENDPOINT = "https://hub-ci.huggingface.co"
 ENDPOINT = os.getenv("HF_ENDPOINT") or (_HF_DEFAULT_STAGING_ENDPOINT if _staging_mode else _HF_DEFAULT_ENDPOINT)
 
-HUGGINGFACE_CO_URL_TEMPLATE = ENDPOINT + "/{repo_id}/resolve/{revision}/{filename}"
+HUGGINGFACE_CO_URL_TEMPLATE = urljoin(ENDPOINT, "/{repo_id}/resolve/{revision}/{filename}")
```

Comment:
`urljoin()` is not necessary here if `ENDPOINT` is already normalized. see my comment here.

Reply:
Yes but I prefer to keep this way when we connect url parts. It can avoid potential issues caused by formats.